lib/active_record/mysql/enum/enum_column_adapter.rb

```diff
@@ -22,16 +22,6 @@ def mysql_column_adapter
       ActiveRecordColumnWithEnums = Enum.mysql_column_adapter
 
       module EnumColumnAdapter
-        def initialize(*args, **kwargs, &block)
-          super
-
-          if type == :enum
-            @default = if @default.present?
-                         @default.to_sym
-                       end
-          end
```

Comment:
Sorry, I think my commit message here might have been unclear. This code is only coercing the default value held in the schema definition. If you instantiate a new object, this default will be pulled and immediately coerced to a symbol on the new object. Example code below - the `severity` column is defined as an enum and the default is `:medium`. This new object has `severity == :medium` immediately after instantiation.

```
  it 'test_coercions' do
    row = EnumerationTestModel.new
    row.color = "blue"
    row.string_field = 'test'
    expect(row.color).to eq(:blue)
    expect(row.severity).to eq(:medium)
    expect(row.save).to be true

    db_row = EnumerationTestModel.find(row.id)
    expect(db_row).not_to be_nil
    expect(db_row.color).to eq(:blue)
    expect(db_row.severity).to eq(:medium)
  end
```

(I've pushed this also to my branch)

There is a discrepancy if you actually pull out the schema definition of the table and query it's default value. It is a string instead of an enum and I did indeed change that test. The removed code will explode on Rails 6.1 and Rails 7.0 if left in and trawling through the rails changes for those two versions when I was working on this, it seems defaults were not meant to be non-standard types like this, but should be string / int / bool etc (symbols not being in the list of expected types). I did not actually change the defaults however in your schema.rb, despite the change that the schema objects in memory have strings assigned.

Unfortunately, I have finished up the project where I was working on this. That project is now running on Rails 7.0, but it is a large Rails monolith (14 years old) and has 600 tables and was working fine with the latest version of this gem  vendored with this pull request. (at least for the use cases we were using this gem for)